### ISSUE TITLE
Regression: Sprite issue with vertical maps

### ISSUE THREAD

**Issue by mbh21** [alekmaul/pvsneslib#255]
This was fixed in earlier 4.x dev branch, but now with 4.1 it is broken again.

When on a vertical map, if you move up or down on the map, then any sprites that scroll off the map, suddenly appear on the opposite side. So, if you move up, any sprites that go off bottom of map, appear at top. If you move down, any sprites that go off top of map, appear at bottom. 

Previously this was fixed with sprite refresh settings, but this fix no longer works.

**Comment by mbh21** [comment]
Please see animated showing moving up and down. Only the middle goomba should be there, the others are ghosts. Attached is also code and sfc file.

![0nogravityobjects](https://github.com/alekmaul/pvsneslib/assets/107952500/e9046d5a-e537-47c9-9155-983824b82c72)
[nogravityobject09.zip](https://github.com/alekmaul/pvsneslib/files/12665006/nogravityobject09.zip)

**Comment by nub1604** [comment]
This behavior is normal on the Snes, take a look in the sprite viewer in Mesen when you moving vertically 
To get around this, just draw the sprite out depending on your camera Y Position.
Try something like this:
`   
    oambuffer[sprnum].oamx = (*goombaox) - x_pos; //  that was easy for X
    
    s16 posY = (*goombaoy) - y_pos;  //define posY on top of your c file, not method local like here
    if (posY > -16 && posY < 240) // play around with these values, in my case it works with a sprite size of 16
    {
        oambuffer[sprnum].oamy = posY; // in if its in range it will draw
    }
    else
    {
        oambuffer[sprnum].oamy = 240; // otherwise draw the sprite it out of screen
    }
    oamDynamic16Draw(sprnum);`

**Comment by mbh21** [comment]
Yes, but this was previously handled by the map engine.

**Comment by alekmaul** [comment]
that's correct, it was handle previously but the code was not correct. Is it possible to have your source code to check how you add your objects in the map?
